pam/internal/adapter/utils.go

```diff
@@ -65,6 +73,46 @@ func isSSHSession(mTx pam.ModuleTransaction) bool {
 	return isSSHSessionValue
 }
 
+func getPamTty(mTx pam.ModuleTransaction) (*os.File, func(), error) {
+	pamTty, err := mTx.GetItem(pam.Tty)
+	if err != nil {
+		return nil, func() {}, err
+	}
+
+	if pamTty == "" {
+		return nil, func() {}, nil
+	}
+
+	tty, err := os.OpenFile(pamTty, os.O_RDWR, 0600)
+	if err != nil {
+		return nil, func() {}, err
+	}
```

Comment:
This will fail if `PAM_TTY` is set to `$DISPLAY`, which according to https://manpages.ubuntu.com/manpages/noble/en/man3/pam_set_item.3.html was done in the past:

> PAM_TTY
>            The terminal name prefixed by [/dev/](file:///dev/) for device files. In the past, graphical X-based
>            applications used to store the $DISPLAY variable here, but with the introduction of
>            PAM_XDISPLAY this usage is deprecated.

Should we print a warning instead?

Reply:
Ah, we even just print a debug message in L105